python/gtsam/preamble/nonlinear.h

```diff
@@ -9,4 +9,5 @@
  * automatic STL binding, such that the raw objects can be accessed in Python.
  * Without this they will be automatically converted to a Python object, and all
  * mutations on Python side will not be reflected on C++.
- */
\ No newline at end of file
+ */
+#include <pybind11/stl.h>
```

Comment:
This is likely the culprit. Automatic STL binding also caused some issues for me in another PR.

Reply:
Makes sense. And how have you solved them?

Reply:
Removed it and used the stl_bind instead